core/src/main/java/com/sequenceiq/cloudbreak/service/environment/EnvironmentService.java

```diff
@@ -177,7 +201,9 @@ public DetailedEnvironmentV4Response get(String environmentName, Long workspaceI
 
     public SimpleEnvironmentV4Response delete(String environmentName, Long workspaceId) {
         Environment environment = getByNameForWorkspaceId(environmentName, workspaceId);
-        delete(environment);
+        LOGGER.debug(String.format("Starting to delete environment [name: %s, workspace: %s]", environment.getName(), environment.getWorkspace().getName()));
+        prepareDeletion(environment);
+        resourceArchivator.archive(environment);
```

Comment:
`resourceArchivator` could be unnecessary and environment.archive() could be called. I think it would be cleaner. (And don't forget to save the changed Environment with the repository.)